lib/moment/collectMoment.ts

```diff
@@ -53,28 +53,27 @@ export async function collectMoment({
     amount
   );
 
-  const calls = [...approveCall, collectCall] as readonly OneOf<
-    Call<unknown, { [key: string]: unknown }>
-  >[];
-  // Send the transaction and wait for receipt using the helper
-  const transaction = await sendUserOperation({
-    smartAccount,
-    network: IS_TESTNET ? "base-sepolia" : "base",
-    calls,
-  });
+  const calls = [...approveCall, collectCall] as OneOf<Call<unknown, { [key: string]: unknown }>>[];
 
-  // Distribute funds from split contract if fundsRecipient is a split
   if (saleConfig.fundsRecipient) {
     const isSplit = await isSplitContract(saleConfig.fundsRecipient as Address, CHAIN_ID);
     if (isSplit) {
-      await distributeSplitFunds({
+      const splitCall = await distributeCall({
```

Comment:
Out of curiosity, why did you decide to implement `automatic` splits distribution in the `UI` Codebase rather than the `cron` codebase?